llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -7972,6 +7972,13 @@ Instruction *InstCombinerImpl::visitFCmpInst(FCmpInst &I) {
   Constant *RHSC;
   if (match(Op0, m_Instruction(LHSI)) && match(Op1, m_Constant(RHSC))) {
     switch (LHSI->getOpcode()) {
+    case Instruction::Select:
+      if (FCmpInst::isEquality(Pred) && match(RHSC, m_AnyZeroFP()) &&
+          (match(LHSI,
+                 m_Select(m_Value(), m_Value(X), m_FNeg(m_Deferred(X)))) ||
+           match(LHSI, m_Select(m_Value(), m_FNeg(m_Value(X)), m_Deferred(X)))))
+        return new FCmpInst(Pred, X, RHSC, "", &I);
```

Comment:
Could you please add some FMF tests?

Reply:
This will drop FMF and the name of the original fcmp inst. How about:
```suggestion
        return replaceOperand(I, 0, X);
```